Tests/NIOSSLTests/NIOSSLIntegrationTest.swift

```diff
@@ -2365,6 +2364,57 @@ class NIOSSLIntegrationTest: XCTestCase {
         XCTAssertEqual(certificates.withLockedValue { $0 }, [NIOSSLIntegrationTest.cert])
     }
 
+    func testExtractingCertificateChainFromCustomVerificationCallback() throws {
```

Comment:
Can I suggest that we also add tests to confirm:

1. The failure path
2. Delayed success (i.e. return success later on, from the main thread).